docs/common/istio-nftables.adoc

```diff
@@ -0,0 +1,333 @@
+
+200
+----
+
+More guidelines:
+https://github.com/istio/istio/tree/master/tools/istio-nftables/pkg#debugging-guidelines[Debugging Guidelines]
+
+==== Upgrade
+
+The migration from iptables backend to nftables backend can be done by upgrading `+Istio+` and `+IstioCNI+` resources. 
+Because the CNI component runs as a cluster singleton, it is recommended to operate and upgrade the CNI component separately from the Istio control plane.
+
+When using Sail Operator, the upgraded `IstioCNI` and `Istio` resources `spec.version` value need to be different from the prior instances. Otherwise, the appended `spec.values` will not be configured.
+
+===== Upgrade with Sail Operator on OpenShift
+
+To upgrade an iptable based Istio service mesh, using the following steps:
+
+[arabic]
+. Check existing `+Istio+` and `+IstioCNI+` resources’ state are Healthy.
+. Upgrade the IstioCNI and Istio control plane with `+spec.values.global.nativeNftables=true+`. 
+More details about the Update Strategy are described in link:../update-strategy/update-strategy.adoc[update-strategy.adoc]. For example,
+
+[source,sh]
+----
+cat <<EOF | kubectl apply -f-
+apiVersion: sailoperator.io/v1
+kind: IstioCNI
+apiVersion: sailoperator.io/v1
+metadata:
+  name: default
+spec:
+  version: master
+  namespace: istio-cni
+  values:
+    global:
+      nativeNftables: true
+EOF
+----
+
+[source,sh]
+----
+cat <<EOF | kubectl apply -f-
+apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  namespace: istio-system
+  updateStrategy:
+    type: InPlace
+  version: master
+  values:
+    global:
+      nativeNftables: true
+EOF
+----
+
+[arabic, start=3]
+. Update the data plane namespace `+test-ns+` by restarting all deployments. For example,
+
+[source,sh]
+----
+kubectl rollout restart deployment -n test-ns
+----
+
+===== Upgrade in Ambient mode
+
+To upgrade `+IstioCNI+` and `+Istio+` resources in Ambient mode, you can set the same feature flag with ambient profile. For example,
+
+[source,sh]
+----
+cat <<EOF | kubectl apply -f-
+apiVersion: sailoperator.io/v1
+kind: IstioCNI
+apiVersion: sailoperator.io/v1
+metadata:
+  name: default
+spec:
+  version: master
+  namespace: istio-cni
+  profile: ambient
+  values:
+    global:
+      nativeNftables: true
+EOF
+----
+
+[source,sh]
+----
+cat <<EOF | kubectl apply -f-
+apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  namespace: istio-system
+  updateStrategy:
+    type: InPlace
+  version: master
+  profile: ambient
+  values:
+    pilot:
+      trustedZtunnelNamespace: ztunnel
+    global:
+      nativeNftables: true
+EOF
+----
+
+[arabic, start=3]
+. Update the data plane namespace `+test-ns+` by restarting all deployments so that new rules can be applied. For example,
+
+[source,sh]
+----
+kubectl rollout restart deployment -n test-ns
```

Comment:
hmm... we should find a way to support re-programming the new rules without having to restart the pods. Did you take a look at the code to understand what it does when config changes and the rules do not match?